src/Umbraco.Web/WebApi/Filters/EnsureUserPermissionForContentAttribute.cs

```diff
@@ -1,143 +1,163 @@
-                    else
-                    {
-                        Guid.TryParse(argument, out Guid key);
-                        nodeId = ApplicationContext.Current.Services.EntityService.GetIdForKey(key, UmbracoObjectTypes.Document).Result;
-                    }
-                }
-                else
-                {
-                    //now we need to see if we can get the property of whatever object it is
-                    var pType = actionContext.ActionArguments[parts[0]].GetType();
-                    var prop = pType.GetProperty(parts[1]);
-                    if (prop == null)
-                    {
-                        throw new InvalidOperationException("No argument found for the current action with the name: " + _paramName);
-                    }
-                    nodeId = (int)prop.GetValue(actionContext.ActionArguments[parts[0]]);                    
-                }
-            }
-            else
-            {
-                nodeId = _nodeId.Value;
-            }
-
-            if (ContentController.CheckPermissions(
-                actionContext.Request.Properties,
-                UmbracoContext.Current.Security.CurrentUser,
-                ApplicationContext.Current.Services.UserService,
-                ApplicationContext.Current.Services.ContentService, 
-                ApplicationContext.Current.Services.EntityService, 
-                nodeId, _permissionToCheck.HasValue ? new[]{_permissionToCheck.Value}: null))
-            {
-                base.OnActionExecuting(actionContext);
-            }
-            else
-            {
-                throw new HttpResponseException(actionContext.Request.CreateUserNoAccessResponse());
-            }
-            
-        }
-
-        
-
-    }
-}
+
+        public EnsureUserPermissionForContentAttribute(int nodeId, char permissionToCheck)
+            : this(nodeId)
+        {
+            _permissionToCheck = permissionToCheck;
+        }
+
+        public EnsureUserPermissionForContentAttribute(string paramName)
+        {
+            if (string.IsNullOrWhiteSpace(paramName)) throw new ArgumentException("Value cannot be null or whitespace.", "paramName");
+
+            _paramName = paramName;
+            _permissionToCheck = ActionBrowse.Instance.Letter;
+        }
+
+        public EnsureUserPermissionForContentAttribute(string paramName, char permissionToCheck)
+            : this(paramName)
+        {
+            _permissionToCheck = permissionToCheck;
+        }
+
+        public override bool AllowMultiple
+        {
+            get { return true; }
+        }
+
+        public override void OnActionExecuting(HttpActionContext actionContext)
+        {
+            if (UmbracoContext.Current.Security.CurrentUser == null)
+            {
+                //not logged in
+                throw new HttpResponseException(System.Net.HttpStatusCode.Unauthorized);
+            }
+
+            var ignoreUserStartNodes = false;
```

Comment:
There should be no changes to this file at all, there's no need since no changes go through the content controller (this is also dangerous to do)